### DISCUSSION TITLE
How to render a `.md` file from a `.erb` template?

### DISCUSSION THREAD

**Discussion by joemasilotti** [bridgetownrb/bridgetown#673] [Q&A] [answered]
The docs on [ERB + Markdown](https://www.bridgetownrb.com/docs/template-engines/erb-and-beyond#markdown) mention:

> Alternatively, you can author a document with a `.md` extension and configure it via `template_engine: erb` to get processed through ERB.

I'm trying to render `src/_partials/_about.md` from `src/index.erb` but the file doesn't seem to be found.

```ruby
# bridgetown.config.yml
template_engine: erb
```

```ruby
# src/index.erb
<%= render "about" %>
```

```ruby
# src/_partials/_about.md
---
template_engine: erb

---

* One
* Two
```

But I keep getting the following error. Restarting the server doesn't help.

```
[Bridgetown]   Conversion error: Bridgetown::Converters::ERBTemplates encountered an error while converting `index.erb'
[Bridgetown]   Exception raised: Errno::ENOENT
[Bridgetown] No such file or directory @ rb_sysopen - /Users/joemasilotti/workspace/projects/masilotti.com/bridgetown/src/_partials/_about.erb
[Bridgetown]                  1: /Users/joemasilotti/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/tilt-2.0.11/lib/tilt/template.rb:236:in `initialize'
[Bridgetown]                  2: /Users/joemasilotti/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/tilt-2.0.11/lib/tilt/template.rb:236:in `open'
[Bridgetown]                  3: /Users/joemasilotti/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/tilt-2.0.11/lib/tilt/template.rb:236:in `read_template_file'
[Bridgetown]                  4: /Users/joemasilotti/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/tilt-2.0.11/lib/tilt/template.rb:85:in `block in initialize'
[Bridgetown]                  5: /Users/joemasilotti/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/tilt-2.0.11/lib/tilt/template.rb:86:in `initialize'
```

I see that it's looking for a `.erb` file but that kind of defeats the purpose, no?

Any ideas on what I'm doing wrong here?

**Comment by jaredcwhite** [comment, marked as answer]
Ah, so partials don't actually support front matter or participate in the rendering pipeline for documents. They're literally just an extended bit of whichever template is in use (ERB, Serbea, etc.).

I think what you'd want to do is creating something like a "snippets" [collection](https://www.bridgetownrb.com/docs/collections), so you could place various markdown files in there. Then you could reference those from within a template rather than using `render`:

```erb
<%= collections.snippets.resources.find { _1.data.slug == "about" }.content %>
```

I'm writing that off the top of my head so if it doesn't work let me know. (You could also DRY that syntax up a bit with a new helper…)

**Reply by joemasilotti** [reply]
That worked, thanks Jared!

```ruby
# plugins/builders/helpers.rb
class Builders::Helpers < SiteBuilder
  def build
    helper :render_snippet do |name|
      site.collections.snippets.resources.find { _1.data.slug == name }.content
    end
  end
end
```

```yaml
# bridgetown.config.yml
collections:
  snippets:
    output: false
```

---

Any tips on converting this YAML to the new `config/initializers.rb` format? I tried with the following but it doesn't seem to line up correctly.

```ruby
# config/initializers.rb
Bridgetown.configure do |config|
  permalink "pretty"
  template_engine "erb"

  config.collections << {
    snippets: {output: false}
  }
end
```

**Reply by jaredcwhite** [reply]
Sure! I think you could just do:

```ruby
Bridgetown.configure do |config|
  permalink "pretty"
  template_engine "erb"

  collections do
    snippets do
      output false
    end
  end
end
```

**Reply by joemasilotti** [reply]
Interestingly enough that doesn't error but also doesn't convert the markdown to HTML.

**Reply by jaredcwhite** [reply]
Geez. 🤦🏻‍♂️ Now I'm wondering if this goes back to the issue of the order of which collection renders versus a different collection. There's not yet a configurable way to define those dependencies.

If it's not too much trouble, could you create a test case repo with this issue and I can try to hammer on in a bit?

**Reply by jaredcwhite** [reply]
In the meantime, you could run it through the `markdownify` helper as a stopgap

**Reply by joemasilotti** [reply]
I'll stick with the `bridgetown.config.yml`, for now!

This [repo/branch](https://github.com/joemasilotti/masilotti.com-bridgetown/tree/collection.md) replicates the issue on the homepage.
